### PR TITLE
registration for strategies and configuration participants

### PR DESCRIPTION
**Pull request by polevgart**

### CODE REVIEW

File: main_play.py [hunk 1]
```diff
@@ -19,25 +19,30 @@ def main():
...
+            exit(1)
+
+        gen_player_names = iter(board.player_names)
+        for strategy_name, cnt in config["fixed_strategies"].items():
+            strategy_cls = strategies_registrant.get_participant(strategy_name)
+            strategy: BaseStrategy = lib_util.load_pickle_or_init(config, strategy_cls)
+            for _ in range(cnt):
+                player_name = next(gen_player_names)
+                named_strategy = copy.deepcopy(strategy)
+                named_strategy.player_namse = player_name
```
Comment (KingAArtur) [inline main_play.py]: `player_namse` typo

**Comment by KingAArtur** [comment]
I have checked this out and it seems to be ok but idk i wanna **MATHEMATICIAN** to look at it

File: strategies/registration/__init__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Registrant:
+    def __init__(self):
+        self._participants = {}
+
+    def registry(self, participant: type):
```
Comment (Mathematician2000) [inline strategies/registration/__init__.py]: register?

File: strategies/registration/__init__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        if name not in self._participants:
+            raise ValueError(f"Participant {name} is not registered")
+
+        return self._participants[name]
+
+
+strategies_registrant = Registrant()
+
+
+def registry_strategy(strategy_cls):
```
Comment (Mathematician2000) [inline strategies/registration/__init__.py]: register?

**Review by Mathematician2000 — CHANGES_REQUESTED** [review]

File: strategies/registration/__init__.py (hunk 3 above)
Comment (polevgart) [reply strategies/registration/__init__.py]: ok

File: strategies/registration/__init__.py (hunk 2 above)
Comment (polevgart) [reply strategies/registration/__init__.py]: ok

**Review by Mathematician2000 — APPROVED** [review]

**Pull request merged**